### ISSUE TITLE
Specify OIDC configuration on Shoot creation/update

### ISSUE THREAD

**Issue by mvladev** [gardener/dashboard#419]
At this moment it's impossible (via the creation UI) to specify the OpenId Connect configuration of the `kube-apiserver`.
Specifying OIDC is key requirement in removing the need for all members of an account to be `cluster-admins`.
Therefore the I would recoomend the following 3 options to be added in the dashboard and visible when the cluster creation / edit dialog is active:

## Specify configuration

In this section of the dialog, the following options should be available:

- **Parameter** `Issuer` 

  **Description** URL of the provider which allows the API server to discover public signing keys. Only URLs which use the `https://` scheme are accepted.  This is typically the provider's discovery URL without a path, for example "https://dex.example.com".  This URL should point to the level below .well-known/openid-configuration 

  **Example** If the discovery URL is `https://dex.example.com/.well-known/openid-configuration`, the value should be `https://dex.example.com` 

  **Required** Yes 

- **Parameter** `Client ID` 

  **Description** A client id that all tokens must be issued for. 

  **Example** `kubernetes` 

  **Required** Yes 

- **Parameter** `Username Claim` 

  **Description** JWT claim to use as the user name. By default `sub`, which is expected to be a unique identifier of the end user. Admins can choose other claims, such as `email` or `name`, depending on their provider. However, claims other than `email` will be prefixed with the issuer URL to prevent naming clashes with other plugins. 

  **Example** `sub` 

  **Required** No 

- **Parameter** `Username Prefix` 

  **Description** Prefix prepended to username claims to prevent clashes with existing names (such as `system:` users). For example, the value `oidc:` will create usernames like `oidc:jane.doe`. If this flag isn't provided and `User Claim` is a value other than `email` the prefix defaults to `( Issuer URL )#` where `( Issuer URL )` is the value of `Issuer`. The value `-` can be used to disable all prefixing. 

  **Example** `oidc:` 

  **Required** No 

- **Parameter** `Groups Claim` 

  **Description** JWT claim to use as the user's group. If the claim is present it must be an array of strings. 

  **Example** `groups`

  **Required** No 

- **Parameter** `Groups Prefix` 

  **Description** Prefix prepended to group claims to prevent clashes with existing names (such as `system:` groups). For example, the value `oidc:` will create group names like `oidc:engineering` and `oidc:infra`. 

  **Example** `oidc:` 

  **Required**  No 

- **Parameter** `Required claim` 

  **Description** A key=value pair that describes a required claim in the ID Token. If set, the claim is verified to be present in the ID Token with a matching value. Repeat this flag to specify multiple claims. 

  **Example** `claim=value` 

  **Required** No 

- **Parameter** `CA Bundle` 

  **Description** Certificate for the CA that signed your identity provider's web certificate.  Defaults to the host's root CAs. 

  **Example** 

  ```
  -----BEGIN CERTIFICATE-----
  MIIBizCCATGgAwIBAgIUH4plk9qwD61FVXgiOTngFU5FeSkwCgYIKoZIzj0EAwIw
  EjEQMA4GA1UEAxMHUm9vdC1DQTAgFw0xNjEwMTEwNTA2MDBaGA8yMTE2MDkxNzA1
  MDYwMFowEjEQMA4GA1UEAxMHUm9vdC1DQTBZMBMGByqGSM49AgEGCCqGSM49AwEH
  A0IABI2CsrAnMGT8P2VGU2MLo5pv86Z74kcV9hgkLJUkSaeNyc1s89w7X5V2wvwu
  iWEJRGm5RoZJausmyZLZEoKEVXejYzBhMA4GA1UdDwEB/wQEAwIBBjAPBgNVHRMB
  Af8EBTADAQH/MB0GA1UdDgQWBBQez+vouP6XzY6mYCb/23YAOxXuLTAfBgNVHSME
  GDAWgBQez+vouP6XzY6mYCb/23YAOxXuLTAKBggqhkjOPQQDAgNIADBFAiBGclts
  vJRM+QMVoV/1L9b+hvhgLIp/OupUFsSOReefIwIhALY06hBklyh8eFwuBtyX2VcE
  8xlVn4/5idUvc3Xv2h9s
  -----END CERTIFICATE-----
  ```

  **Required** No 

- **Parameter** `Signing Algorithms` 

  **Description** Comma-separated list of allowed JOSE asymmetric signing algorithms. JWTs with a 'alg' header value not in this list will be rejected. Values are defined by RFC 7518 https://tools.ietf.org/html/rfc7518#section-3.1. Defaults to `RS256` 

  **Example** `RS256` 

  **Required**  No 

## Inherit Gardener API Server configuration.

This uses the OIDC configuration of the  Gardener Kubernetes API Server. It **MUST** be provided as configuration to the dashboard on installation, otherwise this option should not be displayed or be grayed-out.

There shoud be a warning indicating that if this is chosen, **ALL** users which can be be authenticated to dashboard can be authenticated to the respective Shoot cluster and it's up to the Project Admins to configure `RBAC`.

> Note: I really would like this option to be default (configurable) option. It can improve security a lot.

## Do not configure.

Does nothing. Current behavior.

There shoud be a warning indicating that if this is chosen, the Garden dashboard cannot provide personalized `kubeconfig` to Project Members.

**Comment by vasu1124** [comment]
## Inherit Project configuration.
This uses the OIDC configuration of the shoot's project. It MUST be provided as configuration to the dashboard on the project level, otherwise a link to creating this configuration should be provided.

**Comment by grolu** [comment]
With the new Create Cluster implementation is merged (see PR #416) this will be possible by editing the yaml before creating a shoot. We need to discuss, if we plan to offer such advanced configuration as part of the UI or if it is sufficient to offer a possibility to configure those things in the yaml.
There are also many other things that currently cannot be configured via the UI (e.g. Shoot DNS configuration, API Server flags etc.)

**Comment by grolu** [comment]
Regarding https://github.com/gardener/gardener/issues/1161 - there is nothing that we need to implement in the dashboard, right? If configured, all shoots will get an OIDC config injected by default.
So besides #462 - is there anything else that we can improve in the dashboard?
